### DISCUSSION TITLE
How to "unwrap" union types?

### DISCUSSION THREAD

**Discussion by tutiplain** [ballerina-platform/ballerina-lang#36936] [Q&A] [answered]
Hi all,

  I am new to Ballerina, so please excuse me if this has been asked somewhere before, but at least I can't find a way to do it in the official docs.  If I have a float|error type variable, I cannot use it in math operations.  So far, what I've been doing is something like this:

`float|error myFloat= getFloatSomehow();
 float myRealFloat;
 if myFloat is error {
  //print some output
} else {
  myRealFloat = myFloat;
}
`

It seems wasteful to have a variable declared just for that. Is there a simpler way to do this?

**Comment by SandaruJayawardana** [comment, marked as answer]
Hi @tutiplain,
Math operations are not allowed with error types. So you have to ensure that the variable `myFloat` doesn't contain an error value before using it for math operations. IMO you can do it using either type narrowing or [checking expressions](https://ballerina.io/spec/lang/2022R1/#section_6.33).

```ballerina
float|error myFloat = 12;
    
if myFloat !is error { // using type narrowing
    myFloat = myFloat * 12;
}
```

```ballerina
public function main() returns error? {
    float|error myFloat = 12;
    myFloat = 12 * (check myFloat); // using check-expr
}
```
Does this solve your problem?

**Reply by SandaruJayawardana** [reply]
For general cases, I think you can use type-narrowing feature to unwrap union types.

**Reply by tutiplain** [reply]
Hi, thanks for your replies. I hadn't learned about check expressions yet. I used a function returning a float|error and using check expressions as in the example above to simplify my code a bit. At least now I don't have two variables, a float and a float|error, in my code.

**Comment by SandaruJayawardana** [comment]
FYI, You can use ballerina slack channel also to find answers to your questions, get support or learn how others are using Ballerina. https://ballerina.io/community/#ballerina-slack-community
